### DISCUSSION TITLE
Ship a function to quote regular expressions.

### DISCUSSION THREAD

**Discussion by benjamingr** [nodejs/help#4608] [Ideas]
Hey,

One of the most "obvious" missing features in Node.js regular expressions that every other platform ships is the ability to "escape" them. That is:

```js
var str = prompt("Please enter a string");
str = escape(str);
const re = new RegExp(str, "g"); // handles reg exp special tokens with the replacement.
```

There are multiple packages doing this subtly incorrectly on NPM - `lodash`, `escape-regexp` and a few others.

I've worked [on a TC39 proposal](https://github.com/benjamingr/RegExp.escape) to add this API to the language and it is being considered again - but if the language committee rejects the proposal - the next best place [is the platform](https://github.com/benjamingr/RegExp.escape/issues/43#issuecomment-738950494).

My current plan is:
 - Try to request that the the language committee add this missing feature. Jordan has helpfully volunteered to raise this in the January TC39 meeting and possibly champion the proposal.
 - If TC39 shuts it down - make a PR to add a utility function to Node.js as suggested on that issue.

In regards to other languages and platforms - it seems that some ship this sort of API as part of the language and some do so as part of the platform. 

I am opening this issue here for discussion of the plan.

An alternative suggested by some whatwg members is trying to coordinate this sort of API with them.

**Comment by benjamingr** [comment]
Bikeshedding: `util.escapeRegExpString` which does the correct thing and keeps up to date with RegExp enhancements sound good to me.

**Comment by tniessen** [comment]
Your TC39 proposal seems very reasonable to me! 👍

**Comment by benjamingr** [comment]
TC39 has agreed to promote the proposal to stage 1 - so I'd really rather pursue it there first.

**Comment by jasnell** [comment]
@benjamingr ... any objections to converting this into a discussion rather than keeping it open as an issue?

**Comment by benjamingr** [comment]
@jasnell sure

**Comment by regseb** [comment]
Proposal TC39 [_RegExp Escaping Proposal_](https://github.com/tc39/proposal-regex-escaping#regexp-escaping-proposal) `RegExp.escape()` is in stage 3. This discussion could be re-converted into an issue.

FYI:

- [_Ship RegExp.escape proposal_](https://bugzil.la/1918235) in Firefox 134.
- [_Implement RegExp.escape_](https://issues.chromium.org/353856236) in Chromium.

**Reply by aduh95** [reply]
> This discussion could be re-converted into an issue.

That wouldn't make much sense, there's no action for us, we have to wait for V8 to implement it.
